src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -179,6 +207,177 @@ def upload_stream(
             "sfqid": None,
         }
 
+    def get_file(
+        self,
+        stage_location: str,
+        target_directory: str,
+        options: Dict[str, str] = None,
+    ) -> TableEmulator:
+        if target_directory.startswith("'file://"):
+            target_directory = target_directory[
+                len("'file://") : -1
+            ]  # skip normalized prefix `file:// and suffix `
+        stage_source_dir_path = os.path.join(self._working_directory, stage_location)
+
+        result_df = TableEmulator(
+            columns=GET_RESULT_KEYS,
+            sf_types={
+                "file": ColumnType(StringType(), True),
+                "size": ColumnType(DecimalType(10, 0), True),
+                "status": ColumnType(StringType(), True),
+                "message": ColumnType(StringType(), True),
+            },
+            dtype=object,
+        )
+
+        # looking for a directory or a file
+        if not (
+            os.path.exists(stage_source_dir_path)
+            or os.path.exists(f"{stage_source_dir_path}{StageEntity.FILE_SUFFIX}")
+        ):
+            raise SnowparkSQLException(
+                f"[Local Testing] the file does not exist: {stage_source_dir_path}"
+            )
+
+        if os.path.isfile(f"{stage_source_dir_path}{StageEntity.FILE_SUFFIX}"):
+            list_of_files = [stage_source_dir_path]
+        else:
+            # here we get all the file names with suffix removed so that pattern can match the original names
+            list_of_files = sorted(
+                os.path.splitext(os.path.join(root, file))[0]
+                for root, dirs, files in os.walk(stage_source_dir_path)
+                for file in files
+            )
+
+        pattern = options.get("pattern") if options else None
+
+        for file in list_of_files:
+            file_name = os.path.basename(file)
+            # pattern[1:-1] to remove heading and tailing single quotes
+            if pattern and not re.match(pattern[1:-1], file_name):
+                continue
+            stage_file = f"{file}{StageEntity.FILE_SUFFIX}"
+            shutil.copy(stage_file, os.path.join(target_directory, file_name))
+            file_size = os.path.getsize(stage_file)
+            result_df.loc[len(result_df)] = dict(
+                zip(
+                    GET_RESULT_KEYS,
+                    [
+                        file_name,
+                        file_size,
+                        "DOWNLOADED",
+                        "",
+                    ],
+                )
+            )
+        return result_df
+
+    def read_file(
```

Comment:
this method is moved from the deleted _file_operation.py to consolidate file operations in the stage registry module